Set non-default backend names to lower case

bryanmr opened the pull request:
Summary: Hopefully this makes naming more consistent. Without this change, some tests will fail for plugins since values can be set to upper case in some cases. This should prevent that and make lookup and comparison consistent.

Test Plan: Check the signals. There is no specific test for this, but all tests should pass.

Differential Revision: D32836529



cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

mrshenli reviewed (approved):
Just wanna double check if I understand this correctly. You mean users might do things like `Backend("UCC")`, `Backend("Ucc")`, `Backend("ucc")`, and they should all land in the same behavior? This looks good to me.

bryanmr commented:
> Just wanna double check if I understand this correctly. You mean users might do things like `Backend("UCC")`, `Backend("Ucc")`, `Backend("ucc")`, and they should all land in the same behavior? This looks good to me.

Hey Shen! Thanks for reviewing. Yeah, we have some scenarios where this mattered for UCC and the `test_Backend_enum_class` will actually fail without some changes. This is the least intrusive. We made changes on the UCC side to make case consistent, but that test purposefully isn't consistent.

bryanmr committed:
Set non-default backend names to lower case (#69400)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/69400

Hopefully this makes naming more consistent. Without this change, some tests will fail for plugins since values can be set to upper case in some cases. This should prevent that and make lookup and comparison consistent.

Test Plan: Check the signals. There is no specific test for this, but all tests should pass.

Reviewed By: mrshenli

Differential Revision: D32836529

fbshipit-source-id: b17c7ca8d5c338a4c52750003d06cceb42a5a6d7
